### PR TITLE
Replace symbols which generate invalid identifiers.

### PR DESCRIPTION
**Pull request by rbartlensky**
The grammar for Lua is defining symbols which get translated into invalid Rust identifiers.
For example:
`\( "("` will be translated into `const T_(: u8 = some_number` which is an invalid identifier.

This branch makes sure that all symbols are encoded as strings which are translated into valid identifiers.

### CODE REVIEW

File: lua5_3/lua5_3.l [hunk 1]
```diff
@@ -22,42 +22,42 @@ true "TRUE"
...
-\} "}"
-\[ "["
-\] "]"
-:: "::"
-; ";"
-: ":"
-, ","
+\+ "PLUS"
+\- "MINUS"
+\* "ASTERIX"
```
Comment (ltratt) [inline lua5_3/lua5_3.l]: I think this should be `STAR`?

File: lua5_3/lua5_3.l [hunk 2]
```diff
@@ -22,42 +22,42 @@ true "TRUE"
...
+>> "GTGT"
+// "FSFS"
+== "EQEQ"
+~= "NOTEQ"
+<= "LE"
+>= "GE"
+< "LT"
+> "GT"
+= "EQ"
+\( "ORB"
```
Comment (ltratt) [inline lua5_3/lua5_3.l]: Probably `LBRACKET`.

File: lua5_3/lua5_3.l [hunk 3]
```diff
@@ -22,42 +22,42 @@ true "TRUE"
...
+\) "CRB"
+\{ "OCB"
```
Comment (ltratt) [inline lua5_3/lua5_3.l]: Probably `LCURLY`.

File: lua5_3/lua5_3.l [hunk 4]
```diff
@@ -22,42 +22,42 @@ true "TRUE"
...
 ("(\\[abfnrtvz\\"']|\\x[0-9a-fA-F]|\\[0-9][0-9]?[0-9]?|\\u\{[0-9a-fA-F]+\}|.)*?")|('(\\[abfnrtvz\\"']|\\x[0-9a-fA-F]|\\[0-9][0-9]?[0-9]?|\\u\{[0-9a-fA-F]+\}|.)*?') "SHORT_STR"
 (\[\[.*?\]\])|(\[=\[.*?\]=\])|(\[==\[.*?\]==\]) "LONG_STR"
 (0[xX][0-9a-fA-F]+(\.[0-9a-fA-F]+)?([pP]-?[0-9a-fA-F]+)?)|(\.[0-9]+([eE]-?[0-9]+)?)|([0-9]+(\.[0-9]*)?([eE]-?[0-9]+)?) "NUMERAL"
 (--\[=====\[.*?\]=====\])|(--\[====\[.*?\]====\])|(--\[===\[.*?\]===\])|(--\[==\[.*?\]==\])|(--\[=\[.*?\]=\])|(--\[\[.*?\]\])|(--.*?$) ;
 [ \n\r\t]+ ;
 #!.*?$ ;
-\. "."
-\.\. ".."
-\.\.\. "..."
+\. "D"
```
Comment (ltratt) [inline lua5_3/lua5_3.l]: `DOT`?

**Comment by ltratt** [comment]
A few minor comments on naming, but I think they're simple enough.

File: lua5_3/lua5_3.l (hunk 4 above)
Comment (rbartlensky) [reply lua5_3/lua5_3.l]: Should `..` be renamed to `DOTDOT` or is `DD` fine?

**Comment by rbartlensky** [comment]
Are the names of `[` and `]` fine? Or should I rename those as well?

File: lua5_3/lua5_3.l (hunk 4 above)
Comment (ltratt) [reply lua5_3/lua5_3.l]: `DOTDOT` sounds right to me.

**Comment by ltratt** [comment]
`LSQUARE` sounds right.

**Comment by rbartlensky** [comment]
Ready for re-review

**Comment by ltratt** [comment]
Looks good! Please squash.

**Comment by rbartlensky** [comment]
Done. Thank you for the review!

**Review by ltratt — APPROVED** [review]

**Pull request merged**